pylint/checkers/design_analysis.py

```diff
@@ -266,9 +266,16 @@ def _get_parents(
         parent = to_explore.pop()
         if parent.qname() in ignored_parents:
             continue
-        parents.add(parent)
-        to_explore.extend(parent.ancestors(recurs=False))
-    return parents
+        if parent not in parents:
+            yield parent
```

Comment:
I think a comment explaining why we do this might be helpful for future reference